lua/compe/matcher.lua

```diff
@@ -12,13 +12,18 @@ Matcher.match = function(context, source, items)
   for i, item in ipairs(items) do
     item.index = i
 
-    local word = item.original_word
+    local word = nil
     if #input > 0 then
-      if item.filter_text and #item.filter_text > 0 then
-        if Character.match(string.byte(input, 1), string.byte(item.filter_text, 1)) then
-          word = item.filter_text
+      for _, key in ipairs({ 'original_abbr', 'filter_text', 'original_word' }) do
```

Comment:
@hrsh7th It doesn't fix #375 for me , because in my case none of `original_abbr`, `filter_text` or `original_word` are matching the prefix, so it falls back on the last one in the list: `original_word`.

If I set `filter_text` as the last item though, it fixes the issue:


```suggestion
      for _, key in ipairs({ 'original_abbr', 'original_word', 'filter_text' }) do
```

Reply:
Ah! Thanks. It's my bad.

I will consider it more deeply...

(I want to prefer `prefix matched abbr` as first priority)

Reply:
That's fine, if you do this fix it works for me, but if you want to do it differently, let me know so I can test again.